cmd/bootstrap-peer-server.go

```diff
@@ -253,5 +238,14 @@ func newBootstrapRESTClient(endpoint Endpoint) (*bootstrapRESTClient, error) {
 		return nil, err
 	}
 
-	return &bootstrapRESTClient{endpoint: endpoint, restClient: restClient, connected: 1}, nil
+	restClient.HealthCheckFn = func() bool {
+		ctx, cancel := context.WithTimeout(GlobalContext, restClient.HealthCheckTimeout)
+		respBody, err := restClient.CallWithContext(ctx, bootstrapRESTMethodHealth, nil, nil, -1)
+		xhttp.DrainBody(respBody)
+		cancel()
+		var ne *rest.NetworkError
+		return !errors.Is(err, context.DeadlineExceeded) && !errors.As(err, &ne)
```

Comment:
Maybe it is just easier to do this:
`return err == nil` ?

Reply:
Yeah. I have a hard time seeing how any error could be a success.

Reply:
@vadmeste it is specifically added to ensure that we don't invoke routines for context cancellation errors and any other errors resulting which are not network related.

Reply:
But it would still mean that for some reason the server is unreachable. What else could it be?

Reply:
> @vadmeste it is specifically added to ensure that we don't invoke routines for context cancellation errors and any other errors resulting which are not network related.

okay I think I understood, you want this because otherwise a node will be easily marked offline unnecessarily  and we really don't want this

Reply:
There can be other reasons @klauspost I can't forsee then but err == nil won't be correct.